### ISSUE TITLE
Update to api v2

### ISSUE THREAD

**Issue by dekker7** [mollie/laravel-mollie#41]
Will this library will be updated to the new 2.0 api version of Mollie and https://github.com/mollie/mollie-api-php?

Thanks

**Comment by nickurt** [comment]
Same question here, can we expect api v2-support soon? cc @willemstuursma

**Comment by willemstuursma** [comment]
Thanks guys. It's a bit tricky because we don't have a lot of Laravel experts in house. So it might take us a small while to get this working.

We're willing to hire a freelancer to perform this conversion, so if you are available or know anyone please contact me via email willem@mollie.com. We'd also really like Cashier support.

**Comment by Cannonb4ll** [comment]
I could take care of this, send a e-mail to Willem already 👍

**Comment by sandervanhooft** [comment]
I'm on it! Stay tuned.

**Comment by sandervanhooft** [comment]
@willemstuursma et al
Progress update: Laravel-mollie `v2.0.0-beta1` planned for next week.

Most functionality changes have now been implemented in my temporary [dev fork](https://github.com/sandervanhooft/laravel-mollie/commits/master). Will need to run some final checks and (re)write the readme file and package migration guide before release.

Some major dependency bumps were required. Php 7.1+, Laravel 5.5+ (LTS), Socialite 3.

Best to maintain a v1 branch as well?

**Comment by willemstuursma** [comment]
It would be great if you could keep the min. PHP version on 5.6, or the minimum version supported by Laravel, whichever is higher.

**Comment by sandervanhooft** [comment]
@willemstuursma 
I agree. I'll try to match Laravel 5.5 LTS [minimum requirement](https://github.com/laravel/laravel/blob/3a50f43c3be644f6168bf888af272425efb8f65c/composer.json#L8), PHP 7.0.0.

**Comment by nickurt** [comment]
Yup, LGTM to follow Laravel 5.5 minimum requirements, looking forward to v2.0.0-beta1 next week 👍

**Comment by sandervanhooft** [comment]
@willemstuursma How do you recommend launching the beta?

With a `v2.0.0-beta1` tag on the master branch and a link to the migration guide in the release notes? Or should this happen on a separate branch?

**Comment by sandervanhooft** [comment]
I'm preparing the PR [here]( https://github.com/sandervanhooft/laravel-mollie) btw.

**Comment by willemstuursma** [comment]
I think it is fine to release 2.0.0 straight away, unless you feel uncomfortable with that. Then your approach is fine.

**Comment by willemstuursma** [comment]
On the master is fine, there won't be many changes to v1 in the future. We can create a v1-develop branch when needed.

**Comment by sandervanhooft** [comment]
PR is ready, but Travis is failing somehow. The fork however is all green (same contents, same travis script). I am communicating with Travis support over mail to resolve this issue.

**Comment by sandervanhooft** [comment]
Finally ironed out the Travis build issues! Will release `v2.0.0-beta1 ` Monday.

**Comment by sandervanhooft** [comment]
Just released `v2.0.0-beta1`!

**Comment by sandervanhooft** [comment]
I have scheduled the stable release for next Monday - if all goes well. Fingers crossed!

**Comment by nickurt** [comment]
I tested the `v2.0.0-beta1` tag with my Mollie Connect/oAuth app and following the upgrade notes/guide it seems working fine 👍

It only took some time to figure out the testmode options (changes) for payments
```php
$mollie->payments()->get($transactionId . '&testmode=' . $isDevMode);

=>

$mollie->payments()->get($transactionId, [
    'testmode' => $isDevMode ? "true" : "false"
]);
```

**Comment by sandervanhooft** [comment]
Thanks @nickurt !

I'm a bit confused by your `$testmode` issue, not familiar with it. Can you elaborate a bit on it?

Is this a Laravel-Mollie specific feature that should be documented in the Laravel-Mollie upgrade guide? Or does it relate to the mollie-api-php client?

**Comment by nickurt** [comment]
This is a [specific Connect-only](https://docs.mollie.com/reference/v2/payments-api/create-payment#mollie-connect-oauth-parameters) option, in v1 of the api a boolean was allowed, but it seems by v2 it only accept a "true" or "false" as "string", it looks like the docs.mollie.com is not correct here tho, since it said "boolean" is allowed.

I think I can PR it in the [https://github.com/mollie/api-documentation](api-documentation) repo.

Any reason why it needs this now @willemstuursma, is this due the HAL specification?

**Comment by willemstuursma** [comment]
If it is passed as part of the query string, it must be the string "true" or "false".

**Comment by sandervanhooft** [comment]
Thanks for clearing that up. As I understand it, there's no need to reference this `testMode` change in the Laravel Mollie docs / source.

**Comment by nickurt** [comment]
Correct, I just mentioned b/c I was little bit stuck with it :+1:

**Comment by sandervanhooft** [comment]
Just released v2.0.0 stable! 🎉

Thanks everyone for your input.
